### PR TITLE
Promote some Util functions to be extensions

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/Util.kt [hunk 1]
```diff
@@ -26,24 +26,16 @@ import okio.Buffer
 import okio.BufferedSource
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/Util.kt]: eventually I wanna shrink Util.kt into nothing

File: okhttp/src/main/java/okhttp3/FormBody.kt [hunk 2]
```diff
@@ -27,8 +27,8 @@ class FormBody internal constructor(
   encodedNames: List<String>,
   encodedValues: List<String>
 ) : RequestBody() {
-  private val encodedNames: List<String> = Util.immutableList(encodedNames)
-  private val encodedValues: List<String> = Util.immutableList(encodedValues)
+  private val encodedNames: List<String> = encodedNames.toImmutableList()
+  private val encodedValues: List<String> = encodedValues.toImmutableList()
```
Comment (TangHuaiZhe) [inline okhttp/src/main/java/okhttp3/FormBody.kt]: `List<String>` can be omitted.

File: okhttp/src/main/java/okhttp3/internal/cache/CacheInterceptor.kt [hunk 3]
```diff
@@ -185,8 +185,8 @@ class CacheInterceptor(internal val cache: InternalCache?) : Interceptor {
 
       @Throws(IOException::class)
       override fun close() {
-        if (!cacheRequestClosed && !discard(this, ExchangeCodec.DISCARD_STREAM_TIMEOUT_MILLIS,
-                MILLISECONDS)) {
+        if (!cacheRequestClosed &&
+            !this.discard(ExchangeCodec.DISCARD_STREAM_TIMEOUT_MILLIS, MILLISECONDS)) {
```
Comment (Egorand) [inline okhttp/src/main/java/okhttp3/internal/cache/CacheInterceptor.kt]: nit: `this` looks redundant here

File: okhttp/src/main/java/okhttp3/internal/http1/Http1ExchangeCodec.kt [hunk 4]
```diff
@@ -403,8 +405,8 @@ class Http1ExchangeCodec(
     override fun close() {
       if (closed) return
 
-      if (bytesRemaining != 0L && !Util.discard(this,
-              ExchangeCodec.DISCARD_STREAM_TIMEOUT_MILLIS, MILLISECONDS)) {
+      if (bytesRemaining != 0L &&
+          !this.discard(ExchangeCodec.DISCARD_STREAM_TIMEOUT_MILLIS, MILLISECONDS)) {
```
Comment (Egorand) [inline okhttp/src/main/java/okhttp3/internal/http1/Http1ExchangeCodec.kt]: nit: remove `this`

File: okhttp/src/main/java/okhttp3/internal/http1/Http1ExchangeCodec.kt [hunk 5]
```diff
@@ -466,8 +468,8 @@ class Http1ExchangeCodec(
 
     override fun close() {
       if (closed) return
-      if (hasMoreChunks && !Util.discard(this, ExchangeCodec.DISCARD_STREAM_TIMEOUT_MILLIS,
-              MILLISECONDS)) {
+      if (hasMoreChunks &&
+          !this.discard(ExchangeCodec.DISCARD_STREAM_TIMEOUT_MILLIS, MILLISECONDS)) {
```
Comment (Egorand) [inline okhttp/src/main/java/okhttp3/internal/http1/Http1ExchangeCodec.kt]: nit: remove `this`

File: okhttp/src/main/java/okhttp3/internal/cache/CacheInterceptor.kt (hunk 3 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/cache/CacheInterceptor.kt]: Good catch. Fixed!

File: okhttp/src/main/java/okhttp3/internal/http1/Http1ExchangeCodec.kt (hunk 4 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/http1/Http1ExchangeCodec.kt]: Fixed!

File: okhttp/src/main/java/okhttp3/internal/http1/Http1ExchangeCodec.kt (hunk 5 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/http1/Http1ExchangeCodec.kt]: Fixed!

**Comment by swankjesse** [comment]
Turns out if you use the Convert Parameter to Receiver refactoring you get some `this.` lint to clean up!

**Pull request merged**